sno/fast_import.py

```diff
@@ -87,16 +89,26 @@ def fast_import_tables(
         f"--depth={max_delta_depth}",
     ] + list(extra_cmd_args)
 
+    orig_branch = get_head_branch(repo)
     if header is None:
-        header = generate_header(repo, sources, message)
+        # import onto a temp branch. then reset the head branch afterwards.
+        # this allows us to check the result before updating the orig branch.
+        import_branch = f'refs/heads/{uuid.uuid4()}'
```

Comment:
This code looks like it would be a better workaround than the existing workaround I have for doing imports at detached HEAD - see temporary_branch in init.py
If you fix this code so that it also works when importing at detached HEAD, you could get rid of my workaround, and fast_import.py would have a simpler interface that supports all cases.